### PR TITLE
Feature/core 1524 did documents on substrate

### PR DESCRIPTION
**Pull request by wulfraem**
- [CORE-1524]

[CORE-1524]: https://evannetwork.atlassian.net/browse/CORE-1524

### CODE REVIEW

File: src/did/did.spec.ts [hunk 1]
```diff
@@ -25,27 +25,37 @@ import { expect, use } from 'chai';
...
 
 (useIdentity ? describe : describe.skip)('DID Resolver', function test() {
-  this.timeout(600000);
+  this.timeout(60_000);
   let accounts0Identity: string;
   let accounts0Did: string;
   let runtimes: Runtime[];
+  let vade: Vade;
+  const vadeConfig = {}; // use default config but enable vade
+  const runtimeConfig = { useIdentity: true, vade: vadeConfig };
```
Comment (Tschuck) [inline src/did/did.spec.ts:52]: You introduced some checks within tests (e.g.: `this.skip();` if vade is set), but currently the test in always runned with vade. Should we run it twice? Once with the old and one with the new logic?

File: src/did/did.ts [hunk 2]
```diff
@@ -306,28 +332,56 @@ export class Did extends Logger {
...
+      const privateKey = await this.options.accountStore.getPrivateKey(
+        this.options.signerIdentity.underlyingAccount,
+      );
+
+      // // Make sure did is whitelisted
+      // await this.vade.ensureWhitelisted(
+      //   did,
+      //   privateKey,
+      //   signerIdentityDid,
+      // );
```
Comment (Tschuck) [inline src/did/did.ts:350]: Can be removed?

```suggestion
```

File: src/verifications/verifications.ts [hunk 3]
```diff
@@ -1023,7 +1044,7 @@ export class Verifications extends Logger {
             const linked = await this.options.executor.executeContractCall(
               this.contracts.registry, 'getLink', description.public.identity,
             );
-            if (!(new RegExp(`${subject.substr(2)}$`, 'i')).test(linked)) {
+            if (!(new RegExp(`${subject.substr(2)}$`, 'i')).test(linked) && !(this.options.vade && linked === nullBytes32)) {
```
Comment (Tschuck) [inline src/verifications/verifications.ts:1047]:
```suggestion
            if (!(new RegExp(`${subject.substr(2)}$`, 'i')).test(linked)
              && !(this.options.vade && linked === nullBytes32)) {
```

File: src/did/did.spec.ts [hunk 4]
```diff
@@ -601,4 +650,180 @@ use(chaiAsPromised);
...
+
+    it('can get DID documents', async () => {
+      // console.time('safe');
+      // await Promise.all([...Array(100)].map(() => vade.didResolve(didToResolve)));
+      // console.timeEnd('safe');
+
+      // await vade.sharedInit({ target: '127.0.0.1', signer: 'local' });
+      // console.time('unsave');
+      // await Promise.all([...Array(100)].map(() => vade.sharedDidResolve(didToResolve)));
+      // console.timeEnd('unsave');
```
Comment (S3bb1) [inline src/did/did.spec.ts:748]:
```suggestion
```

can be removed?

File: src/did/did.spec.ts (hunk 1 above)
Comment (wulfraem) [reply src/did/did.spec.ts:52]: `this.skip` will be removed and `vade` will be enabled by default.

**Pull request merged**